All_years/04-Social_Care/Care_Home_data.R

```diff
@@ -0,0 +1,304 @@
+  arrange(chi, sending_location, social_care_id, ch_admission_date, sc_latest_submission) %>%
+  group_by(chi, sending_location, social_care_id, ch_admission_date) %>%
+  # counter for latest submission
+  mutate(latest_submission_counter = pmax(sc_latest_submission != lag(sc_latest_submission), FALSE, na.rm = TRUE)) %>%
+  mutate(sum_latest_submission = cumsum(latest_submission_counter)) %>%
+  mutate(
+    ch_admission_date = if_else(sum_latest_submission == min(sum_latest_submission), ch_admission_date, qtr_start),
+    ch_discharge_date = if_else(sum_latest_submission == max(sum_latest_submission), ch_discharge_date, record_date)
+  ) %>%
+  ungroup()
+
+
+# Compare to Deaths Data ---------------------------------------
+
+deaths_data <- haven::read_sav(get_slf_deaths_path())
+
+# match ch_episode data with deaths data
+matched_deaths_data <- ch_episode %>%
+  left_join(deaths_data, by = "chi") %>%
+  # compare discharge date with NRS and CHI death date
+  group_by(chi, sending_location, social_care_id, ch_admission_date) %>%
+  # if either of the dates are 5 or fewer days before discharge
+  # adjust the discharge date to the date of death
+  # corrects most cases of ‘discharge after death’
+  mutate(
+    dis_after_death = ymd(death_date) < ymd(ch_discharge_date) - days(5) | ymd(death_date) == ymd(ch_discharge_date) - days(5),
+    dis_after_death = replace_na(dis_after_death, FALSE)
+  ) %>%
+  mutate(ch_discharge_date = if_else(dis_after_death == TRUE, death_date, ch_discharge_date)) %>%
+  # remove any episodes where discharge is now before admission, i.e. death was before admission
+  mutate(
+    dis_before_adm = ch_discharge_date < ch_admission_date,
+    dis_before_adm = replace_na(dis_before_adm, FALSE)
+  ) %>%
+  filter(dis_before_adm == FALSE) %>%
+  ungroup()
+
+
+# Continuous Care Home Stays ---------------------------------------
+
+# stay will be continuous as long as the admission date is the next day or earlier than the previous discharge date
+
+ch_markers <- matched_deaths_data %>%
+  # ch_chi_cis
+  group_by(chi, sending_location, social_care_id) %>%
+  mutate(continuous_stay_chi = pmax(ch_admission_date != lag(ch_discharge_date) + days(1), FALSE, na.rm = TRUE)) %>%
+  mutate(ch_chi_cis = cumsum(continuous_stay_chi) + 1) %>%
+  ungroup() %>%
+  # ch_sc_id_cis
+  # uses the social care id and sending location so can be used for episodes that are not attached to a CHI number
+  # This will restrict continuous stays to each Local Authority
+  group_by(social_care_id, sending_location) %>%
+  mutate(continuous_stay_sc = pmax(ch_admission_date != lag(ch_discharge_date) + days(1), FALSE, na.rm = TRUE)) %>%
+  mutate(ch_sc_id_cis = cumsum(continuous_stay_sc) + 1) %>%
+  ungroup()
+
+
+# Outfile ---------------------------------------
+
+outfile <- ch_markers %>%
+  mutate(person_id = paste0(sending_location, "-", social_care_id)) %>%
```

Comment:
Could we have this as a function, please? We'll use it in other scripts and in the future for other datasets.

```r
create_person_id <- function(data, type = c("SC")) {
if (type == "SC") {
mutate(data, person_id = paste0(sending_location, "-", social_care_id))
} else {
data
}
}